src/layers/transformer/embeddings.rs

```diff
@@ -2,55 +2,222 @@ use candle_core::{Module, ModuleT, Tensor};
+        self.embedding_dropout = embedding_dropout;
+        self
+    }
+
+    /// Set the layer normalization to apply to the embeddings.
+    ///
+    /// Default: `Identity`.
+    pub fn embedding_layer_norm(mut self, embedding_layer_norm: Box<dyn BuildModule>) -> Self {
+        self.embedding_layer_norm = embedding_layer_norm;
+        self
+    }
+
+    /// Set the width of the embeddings.
+    ///
+    /// Default: `768`.
+    pub fn embedding_width(mut self, embedding_width: usize) -> Self {
+        self.embedding_width = embedding_width;
+        self
+    }
+
+    /// Set the width of the transformer.
+    ///
+    /// The embedding layer will use a projection if the hidden width is
+    /// not equal to the embedding width.
+    ///
+    /// Default: `768`.
+    pub fn hidden_width(mut self, hidden_width: usize) -> Self {
+        self.hidden_width = hidden_width;
+        self
+    }
+
+    /// Set the number of position embeddings.
+    ///
+    /// Default: `None`.
+    pub fn n_positions(mut self, n_positions: Option<usize>) -> Self {
+        self.n_positions = n_positions;
+        self
+    }
+
+    /// Set the vocabulary size (number of embeddings).
+    ///
+    /// Default: `30000`.
+    pub fn n_pieces(mut self, n_pieces: usize) -> Self {
+        self.n_pieces = n_pieces;
+        self
+    }
+
+    /// Set the token type vocabulary size (number of token type embeddings).
+    ///
+    /// Default: `None`.
+    pub fn n_types(mut self, n_types: Option<usize>) -> Self {
+        self.n_types = n_types;
+        self
+    }
+
+    /// Set the dropout to apply after embedding projection.
+    ///
+    /// Default: `Identity`.
+    pub fn projection_dropout(mut self, projection_dropout: Box<dyn BuildModule>) -> Self {
+        self.projection_dropout = projection_dropout;
+        self
+    }
+
+    /// Set the layer normalization to apply to the projection layer.
+    ///
+    /// Default: `Identity`.
+    pub fn projection_layer_norm(mut self, projection_layer_norm: Box<dyn BuildModule>) -> Self {
+        self.projection_layer_norm = projection_layer_norm;
+        self
+    }
 }
 
-impl Default for EmbeddingLayerDropouts {
+impl Default for TransformerEmbeddingsConfig {
     fn default() -> Self {
-        EmbeddingLayerDropouts {
-            embed_output_layer_dropout: Box::new(Identity),
-            proj_output_layer_dropout: Box::new(Identity),
+        Self {
+            embedding_dropout: Box::new(Identity),
+            embedding_layer_norm: Box::new(Identity),
+            embedding_width: 768,
+            hidden_width: 768,
+            n_positions: None,
+            n_pieces: 30000,
+            n_types: None,
+            projection_dropout: Box::new(Identity),
+            projection_layer_norm: Box::new(Identity),
         }
     }
 }
 
 /// Errors for transformer embeddings.
 #[derive(Debug, Snafu)]
 pub enum TransformerEmbeddingsError {
+    #[snafu(display("Cannot build dropout module"))]
```

Comment:
```suggestion
    #[snafu(display("Cannot build dropout"))]
```
For consistency, as seen elsewhere.